pluto/pluto.go

```diff
@@ -20,83 +25,148 @@ var (
 	ErrOverflow = "error: Server sent extra bytes"
 )
 
-type worker struct {
-	begin uint64
-	end   uint64
-	url   *url.URL
+// Stats is returned in a channel by Download function every 250ms and contains details like Current download speed in bytes/sec and amount of data Downloaded
+type Stats struct {
+	Downloaded uint64
+	Speed      uint64
+	Size       uint64
 }
 
-// FileMeta contains information about the file like it's Size, Name and if the server supports multipart downloads
-type FileMeta struct {
-	u                  *url.URL
+type fileMetaData struct {
+	url                *url.URL
 	Size               uint64
 	Name               string
 	MultipartSupported bool
 }
 
-// Stats is returned in a channel by Download function every 250ms and contains details like Current download speed in bytes/sec and amount of data Downloaded
-type Stats struct {
-	Downloaded uint64
-	Speed      uint64
-}
-
-// Config contains all the details that Download needs.
+// Pluto contains all the details that Download needs.
 // Connections is the number of connections to use to download a file
 // Verbose is to enable verbose mode.
 // Writer is the place where downloaded data is written.
 // Headers is any header that you may need to send to download the file.
 // StatsChan is a channel to which Stats are sent, It can be nil or a channel that can hold data of type *()
-type Config struct {
-	Connections uint
-	Verbose     bool
-	Headers     []string
-	Writer      io.WriterAt
-	Meta        *FileMeta
+type Pluto struct {
 	StatsChan   chan *Stats
+	Finished    chan struct{}
+	connections uint
+	verbose     bool
+	headers     []string
 	downloaded  uint64
+	metaData    fileMetaData
+	startTime   time.Time
+	saveFile    string
```

Comment:
`saveFile` should not be a string and `pluto` must not create a file itself as it restricts pluto's use as a library. Please keep it like it was before because that was more flexible.

Reply:
makes sense, I'll change that